diff --git a/plugins/tinyalsa/test/Makefile.am b/plugins/tinyalsa/test/Makefile.am
index 40cc35f..cd01c19 100644
--- a/plugins/tinyalsa/test/Makefile.am
+++ b/plugins/tinyalsa/test/Makefile.am
@@ -25,6 +25,12 @@ agmcap_SOURCES  = agmcap.c
 agmcap_la_CFLAGS = $(AM_CFLAGS)
 agmcap_LDADD    = -ltinyalsa libagmmixer.la
 
+bin_PROGRAMS += agmhostless
+agmhostless_SOURCES  = agmhostless.c
+
+agmhostless_la_CFLAGS = $(AM_CFLAGS)
+agmhostless_LDADD    = -ltinyalsa libagmmixer.la
+
 # install xml files under /etc
 root_etcdir      = "/etc"
 root_etc_SCRIPTS = backend_conf.xml
