From bbd06f5b1689d2e4f4f99b7e3919c040da42bc2d Mon Sep 17 00:00:00 2001
From: Miaoqing Pan <quic_miaoqing@quicinc.com>
Date: Wed, 25 Sep 2024 15:36:48 +0530
Subject: [PATCH] PENDING: networkmanager: allow access tmpfs

Upstream-Status: Pending

Signed-off-by: Miaoqing Pan <quic_miaoqing@quicinc.com>
---
 policy/modules/services/networkmanager.te | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/policy/modules/services/networkmanager.te b/policy/modules/services/networkmanager.te
index 8535832e7..4c8853e57 100644
--- a/policy/modules/services/networkmanager.te
+++ b/policy/modules/services/networkmanager.te
@@ -199,7 +199,7 @@ userdom_read_user_certs(NetworkManager_t)
 userdom_write_user_tmp_sockets(NetworkManager_t)
 userdom_dontaudit_use_unpriv_user_fds(NetworkManager_t)
 userdom_dontaudit_use_user_ttys(NetworkManager_t)
-
+init_read_script_tmp_files(NetworkManager_t)
 optional_policy(`
 	avahi_domtrans(NetworkManager_t)
 	avahi_kill(NetworkManager_t)
-- 
2.25.1
